### ISSUE TITLE
Enable PostLocalSGDOptimizer on CUDA tensors

### ISSUE THREAD

**Issue by spetrescu** [pytorch/pytorch#87696]
### 🚀 The feature, motivation and pitch

Hello!

I am trying to leverage the power of the `PostLocalSGDOptimizer` for averaging gradients during distributed training. However, as mentioned in the [docs](https://pytorch.org/docs/stable/distributed.optim.html#), I see that this optimization is not supported on CUDA tensors.

I think that enabling `PostLocalSGDOptimizer` for CUDA tensors would benefit tremendously the community, and thus, I was wondering: is this currently being worked on?

**Comment by spetrescu** [comment]
Hi! I was wondering if there are any updates on this. :)

**Comment by spetrescu** [comment]
Hi, I was wondering if you would be able to provide a starting point to tackle the issue? I would like to work on it, if possible.
